### PR TITLE
Fix for wrong connection count on remote nodes

### PR DESCRIPTION
**Pull request by HarshDaryani896**
**Description**
`Issue:` Even after setting `CONNECTION_POOL_REMOTE_SIZE` to zero, connections were getting created to remote nodes. Number of connections on REMOTE nodes were equal to Number of connections on LOCAL nodes.

On some investigation we found out that the private localDc variable of `YugabyteDefaultLoadBalancingPolicy` class [child class] was hiding the localDc variable of the `BasicLoadBalancingPolicy` class [parent class], due to which the driver was not properly identifying Local AND Remote nodes. It was considering all Up nodes as Local.

**Fix**
Removed the private localDc from `YugabyteDefaultLoadBalancingPolicy` and made the localDc variable in `BasicLoadBalancingPolicy` protected.
Also did this change for another private member variable of `YugabyteDefaultLoadBalancingPolicy` class: `distanceReporter`

**Note**
This PR also includes some formatting changes in [PartitionAwarePolicy.java](https://github.com/yugabyte/cassandra-java-driver/pull/37/files#diff-7fe2d77df614e4e4765e2a79fc842386aceea6a85d409b1a76cb33adad24093f)

**Testing**

1. Tested locally using a sample app to verify remote and local node connections.
2. [nosqlbench](https://github.com/yugabyte/nosqlbench)  test runs with different consistency level and local datacenter specifications has been done, and load routing behaviour is as expected.
3. Tests in [ycql-4.x-tests repo](https://github.com/yugabyte/ycql-4.x-tests): 7 Tests, 5 Passed, 2 Errors in `TestDMLRoutingLocality`(these occur with 4.15.0-yb-1 also).
4. Tests in [Yugabyte-db repo](https://github.com/yugabyte/yugabyte-db/tree/master/java/yb-cql-4x/src/test/java/org/yb/loadtest): 13 Tests, all Passed.

### CODE REVIEW

**Review by kneeraj — APPROVED** [review]

File: core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/YugabyteDefaultLoadBalancingPolicy.java [hunk 1]
```diff
@@ -45,7 +45,6 @@ public class YugabyteDefaultLoadBalancingPolicy extends BasicLoadBalancingPolicy
       LoggerFactory.getLogger(YugabyteDefaultLoadBalancingPolicy.class);
 
   private volatile DistanceReporter distanceReporter;
```
Comment (ashetkar) [inline core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/YugabyteDefaultLoadBalancingPolicy.java]: We should also check how other `private` members will be affected, for example `distanceReporter` is one.

Comment (HarshDaryani896) [reply core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/YugabyteDefaultLoadBalancingPolicy.java]: Made the changes for `distanceReporter`, I do not see any other private member variable in `YugabyteDefaultLoadBalancingPolicy` (other than LOG).

**Comment by ashetkar** [comment]
Also, please run the formatting command and commit the changes
```
mvn com.coveo:fmt-maven-plugin:format
```

**Review by kneeraj — CHANGES_REQUESTED** [review]
Should get a clean run of nosqlbench with correct routing behaviour before merging.

**Comment by HarshDaryani896** [comment]
> Should get a clean run of nosqlbench with correct routing behaviour before merging.

@kneeraj nosqlbench run with different consistency level and local datacenter specifications has been done, and load routing behaviour is as expected. cc: @ashetkar

**Comment by ashetkar** [comment]
@HarshDaryani896 Let's get this PR cleaned of all the debugging logs and merge it, if we have successfully completed all the testing.
We can work on debugging logs on another branch.
Also, please mention the test case which was failing earlier but is fixed by your change, for reference.

File: core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java [hunk 2]
```diff
@@ -186,6 +222,9 @@ public UpHostIterator(
       // shuffled.
       if (getConsistencyLevel() == ConsistencyLevel.YB_CONSISTENT_PREFIX) {
         // this is to be performed in the local copy
+        PreparedStatement ps = statement.getPreparedStatement();
+        String q = ps == null ? "" : ps.getQuery();
+        LOG.trace("Shuffling the nodes since CL is YB_CONSISTENT_PREFIX for query = {}", q);
```
Comment (ashetkar) [inline core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java]: Please change to debug

File: core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java [hunk 3]
```diff
@@ -290,6 +329,9 @@ public static int getKey(BoundStatement stmt) {
     // are literal
     // constants.
     if (hashIndexes == null || hashIndexes.isEmpty()) {
+      LOG.info(
```
Comment (ashetkar) [inline core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java]: Please change to debug

File: core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java [hunk 4]
```diff
@@ -306,13 +348,17 @@ public static int getKey(BoundStatement stmt) {
       }
       channel.close();
 
-      return getKey(bs.toByteArray());
+      int returnValue = getKey(bs.toByteArray());
+      if (returnValue < 0) {
+        LOG.info("getKey(): Returning negative hash {} for {}", returnValue, pstmt.getQuery());
```
Comment (ashetkar) [inline core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java]: Change this to warn?

**Review by ashetkar — COMMENTED** [review]
Also, please add a log at debug/info 1) when metadata is refreshed and 2) when leader is not found for a particular tablet.

**Review by kneeraj — APPROVED** [review]
lgtm

**Pull request merged**